rules/namespace.go

```diff
@@ -130,6 +101,25 @@ func newNamespace(namespace *schema.Namespace) (Namespace, error) {
 	}, nil
 }
 
+// NamespaceView is a human friendly representation of a namespace at a single point in time.
+type NamespaceView struct {
+	Name              string
+	ForRuleSetVersion int
+	Tombstoned        bool
+}
+
+func (n Namespace) namespaceView(snapshotIdx int) (*NamespaceView, error) {
+	if snapshotIdx < 0 || snapshotIdx > len(n.snapshots) {
```

Comment:
Should probably be `snapshotIdx >= len(n.snapshots)`? Could you add a unit test to test that case?